### ISSUE TITLE
Logging realm activity

### ISSUE THREAD

**Issue by TiagoVeloso** [realm/realm-cocoa#2617]
Is it possible to log Realm activity (queries, insertions, etc) with something like CocoaLumberjack?

Some times while debugging it would be nice to see what queries are going on on Realm. Like what values are being inserted and what not.

**Comment by kylebshr** [comment]
I haven't used it a lot, but I think you're looking for `RBQRealmChangeLogger`.

**Comment by jpsim** [comment]
> I haven't used it a lot, but I think you're looking for `RBQRealmChangeLogger`.

Not quite. That's the sub-component of https://github.com/Roobiq/RBQFetchedResultsController responsible for tracking mutating operations manually (insertions, moves, updates, etc.). It doesn't log anything to the console or other logs.

If you wish to log some of Realm's internal operations while debugging, I'd recommend using [symbolic breakpoints in Xcode](https://developer.apple.com/library/ios/recipes/xcode_help-breakpoint_navigator/articles/adding_a_symbolic_breakpoint.html).

Or if you use CocoaPods, you could edit Realm's source to insert logs and other instrumentation against your app. Just remember to revert those when you ship!

**Comment by TiagoVeloso** [comment]
Okay, It would be nice to have something that does that.

Perhaps some time down the road.

**Comment by jpsim** [comment]
> Okay, It would be nice to have something that does that.

Agreed. Core Data's logging was invaluable whenever I've had to debug it, so I've made this ticket a feature request so we can track this.

**Comment by i-schuetz** [comment]
As written in https://github.com/realm/realm-cocoa/issues/4474 I think this would be a very helpful addition.

Re-posting this [small excerpt](http://stackoverflow.com/a/34617082/930450) from [Slick](http://slick.lightbend.com/), something like this would be great.
![screen shot 2016-12-21 at 22 21 59](https://cloud.githubusercontent.com/assets/1381744/21425304/a90276e4-c848-11e6-8a63-3ace32ed6001.png)

**Comment by johndpope** [comment]
To chime in here - we experienced unexpected results by incorrectly using .all and passing in an empty array which blew away existing data. 
-                realm.add(data, update: .all)
+                realm.add(data, update: .modified)
If it was my framework and people were using this - I'd like to log out a print saying
**WARNING - destroying all data.**
